runtime-attach/src/main/java/io/opentelemetry/contrib/attach/AgentFileLocator.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.contrib.attach;
+
+import io.opentelemetry.javaagent.OpenTelemetryAgent;
+import java.io.File;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.security.CodeSource;
+
+class AgentFileLocator {
+
+  static File locateAgentFile() {
+    CodeSource codeSource = OpenTelemetryAgent.class.getProtectionDomain().getCodeSource();
+
+    if (codeSource == null) {
+      throw new IllegalStateException("could not get agent jar location");
+    }
+
+    URL codeSourceLocation = codeSource.getLocation();
+    try {
+      File javaagentFile = new File(codeSourceLocation.toURI());
```

Comment:
Is there anything to worry about for a fat jar?

Reply:
I haven't try, but I believe `io.opentelemetry.javaagent.OpenTelemetryAgent#verifyJarManifestMainClassIsThis` will "take care" of it. Meaning that fat jar will not be supported for now.